nncf/openvino/quantization/quantize_model.py

```diff
@@ -57,6 +57,13 @@
 TTensor = TypeVar("TTensor")
 
 
+def get_actual_subset_size(dataset, subset_size):
+    dataset_length = dataset.get_length()
+    if dataset_length:
+        return min(dataset_length, subset_size)
+    return subset_size
```

Comment:
if `__len()__` is not implemented, the actual subset size should be obtained as:
```python
actual_subset_size = 0
for data in islice(calibration_dataset.get_inference_data(), subset_size):
    actual_subset_size += 1
```

Reply:
Sorry for the late reply. I found that the suggested changes do not work because the `calibration_dataset.get_inference_data()` iterator is consumed before dump_parameters. As a result, `actual_subset_size` ends up being zero. Could you please provide some suggestions on how to handle this issue?
Thank you very much! @l-bat

Reply:
@awayzjj after discussing within the team, we decided **not** to add `actual_subset_size` to `rt_info` because `rt_info` is intended to contain quantization parameters provided to algorithms. If the provided dataset is shorter than `subset_size`, we should display a warning during the statistics collection process: https://github.com/openvinotoolkit/nncf/blob/408c67dd67fadd20b4d76ce4bc20c2b215258720/nncf/common/tensor_statistics/aggregator.py#L76